File: info/1. Алгоритмы/4. Комбинаторика/8. Генерация сочетаний в лексикографическом порядке/code_sample/Fortran/sample.f90
program sample
implicit none

call print_combination(3,5)

contains

subroutine print_combination(k,n)
   implicit none
   integer,intent(in)::k,n
   integer::comb(1:k)
   integer::i,m
   comb = [(i,i=1,k)]
   do
      write(*,*) comb
      m = -1
      do i = k,1,-1
         if(comb(i)<n-k+i) then
            comb(i) = comb(i) + 1
            m = i
            exit
         end if
      end do
      if (m==-1) then
         exit
      end if
      do i = m+1,k
         comb(i) = comb(i-1) + 1
      end do
   end do
end subroutine print_combination

end program sample

File: info/1. Алгоритмы/4. Комбинаторика/8. Генерация сочетаний в лексикографическом порядке/code_sample/Python/sample.py
def print_combination(k, n):
    comb = [i+1 for i in range(k)]
    while True:
        print(comb)
        m = -1
        for i in range(k-1, -1, -1):
            if comb[i] <= n-k+i:
                m = i
                comb[i] += 1
                break
        if m == -1:
            break
        for i in range(m+1, k):
            comb[i] = comb[i-1]+1


print_combination(3, 5)
